features/FEATURE_BLE/source/generic/GenericSecurityManager.cpp

```diff
@@ -653,14 +653,20 @@ ble_error_t GenericSecurityManager::generateOOB(
     /* Secure connections. Avoid generating if we're already waiting for it.
      * If a local random is set to 0 it means we're already calculating. */
     if (!is_all_zeros(_oob_local_random)) {
-        status = _pal.generate_secure_connections_oob();
+        /* save the current values in case the call to
+         * generate_secure_connections_oob fails */
+        address_t orig_local_address = _oob_local_address;
+        oob_lesc_value_t orig_local_random = _oob_local_random;
+
+        _oob_local_address = *address;
+        /* this will be updated when calculation completes,
+         * a value of all zeros is an invalid random value */
+        set_all_zeros(_oob_local_random);
 
-        if (status == BLE_ERROR_NONE) {
-            _oob_local_address = *address;
-            /* this will be updated when calculation completes,
-             * a value of all zeros is an invalid random value */
-            set_all_zeros(_oob_local_random);
-        } else if (status != BLE_ERROR_NOT_IMPLEMENTED) {
+        status = _pal.generate_secure_connections_oob();
+        if (status != BLE_ERROR_NONE && status != BLE_ERROR_NOT_IMPLEMENTED) {
```

Comment:
you need to now: return status below

Reply:
status is returned on line 670 for the failure case.  Do you mean you want BLE_ERROR_NOT_IMPLEMENTED to be able to be propagated back to the user?  This was not the case originally.

Reply:
nevermind, this is a separate bug since the caller doesn't handle the case and needs fixing too - will fix in separate PR